test/e2e/httpproxy/httpproxy_test.go

```diff
@@ -428,3 +456,20 @@ descriptors:
 		})
 	})
 })
+
+// httpProxyValid returns true if the proxy has a .status.currentStatus
+// of "valid".
+func httpProxyValid(proxy *contourv1.HTTPProxy) bool {
+
+	if proxy == nil {
+		return false
+	}
+
+	if len(proxy.Status.Conditions) == 0 {
+		return false
+	}
+
+	cond := proxy.Status.GetConditionFor("Valid")
+	return cond.Status == "True"
+
+}
```

Comment:
this function has moved into test/e2e/framework.go (`HTTPProxyValid`), so let's drop this copy and use that one instead.

Reply:
I can do this along with #4327 as a follow-up since it's just cleanup.